src/run.ts

```diff
@@ -186,6 +190,82 @@ async function destroy_stage(options: {
     });
   }
 
+  const stacks = await getAllStacksForStage(
+    `${process.env.REGION_A}`,
+    options.stage,
+    filters
+  );
+
+  const protectedStacks = stacks
+    .filter((i) => i.EnableTerminationProtection)
+    .map((i) => i.StackName);
+
+  if (protectedStacks.length > 0) {
+    console.log(
+      `We cannot proceed with the destroy because the following stacks have termination protection enabled:\n${protectedStacks.join(
+        "\n"
+      )}`
+    );
+  } else {
+    console.log(
+      "No stacks have termination protection enabled. Proceeding with the destroy."
+    );
+  }
+
+  const templates = await getCloudFormationTemplatesForStage(
+    `${process.env.REGION_A}`,
+    options.stage,
+    filters
+  );
+
+  const resourcesToCheck = [
+    {
+      templateKey: `database-${options.stage}`,
+      resourceKey: "FormAnswersTable",
+    },
+    {
+      templateKey: `database-${options.stage}`,
+      resourceKey: "FormQuestionsTable",
+    },
+    {
+      templateKey: `database-${options.stage}`,
+      resourceKey: "FormTemplatesTable",
+    },
+    { templateKey: `database-${options.stage}`, resourceKey: "FormsTable" },
+    {
+      templateKey: `database-${options.stage}`,
+      resourceKey: "StateFormsTable",
+    },
+    { templateKey: `database-${options.stage}`, resourceKey: "StatesTable" },
+    { templateKey: `database-${options.stage}`, resourceKey: "AuthUserTable" },
+    {
+      templateKey: `ui-${options.stage}`,
+      resourceKey: "CloudFrontDistribution",
+    },
+    { templateKey: `ui-auth-${options.stage}`, resourceKey: "CognitoUserPool" },
+  ];
```

Comment:
You'd have to change some other code I'm sure but I wonder if you like this stripped down version of this same data:
```js
const resourcesToCheck = {
  `database-${options.stage}`: "FormAnswersTable",
  `database-${options.stage}`: "FormQuestionsTable",
  `database-${options.stage}`: "FormTemplatesTable",
  `database-${options.stage}`: "FormsTable"
  `database-${options.stage}`: "StateFormsTable",
  `database-${options.stage}`,: "StatesTable"
  `database-${options.stage}`:  "AuthUserTable"
  `ui-${options.stage}`: "CloudFrontDistribution",
  `ui-auth-${options.stage}`: "CognitoUserPool" 
}
```